errbot/backends/irc.py

```diff
@@ -190,7 +190,7 @@ def join(self, username=None, password=None):
         if username is not None:
             log.debug("Ignored username parameter on join(), it is unsupported on this back-end.")
         if password is None:
-            password = ""
+            password = str()
```

Comment:
Not much really.
The options are `str()` OR `""` followed by a `nosec` comment and I'm trying to avoid using those.